src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/BalloonPopup.java

```diff
@@ -61,26 +66,42 @@ public void recalculateBounds() {
     var maxSize = getMaximumSize();
     var prefSize = getPreferredSize();
 
-    int popupWidth = Integer.min(maxSize.width, prefSize.width);
-    int popupHeight = prefSize.height;
+    var popupWidth = Integer.min(maxSize.width, prefSize.width);
+    var popupHeight = prefSize.height;
```

Comment:
Nit: I prefer primitive type names over `var`. `var` is fine for other types.

Reply:
Makes sense, I'll change this